src/services/helpers.ts

```diff
@@ -98,7 +104,7 @@ export const getNextExercise = async (profession: Discipline): Promise<NextExerc
   }
   const progress = await getExerciseProgress()
   const firstUnfinishedDisciplineId = leafDisciplineIds.find(
-    id => getDoneExercisesByProgress(id, progress) < EXERCISES.length
+    id => getWellEnoughDoneExercises(id, progress) < EXERCISES.length
   )
 
   if (!firstUnfinishedDisciplineId) {
```

Comment:
Blow in line 123 you also need to adjust the line 
`const nextExerciseKey = EXERCISES.find(exercise => disciplineProgress[exercise.key] === undefined)`

to use your new funtion: `const nextExerciseKey = getWellEnoughDoneExercises(firstUnfinishedDisciplineId, progress)`. Otherwise here it would be ignored if the exercise is done well enough.